dashboard/app/models/census/state_cs_offering.rb

```diff
@@ -117,24 +121,39 @@ class Census::StateCsOffering < ApplicationRecord
     VA
   ).freeze
 
+  # For a few states, we already placed a 2017-2018 CSV on S3 using their V1 format,
+  # but they are now getting an update mid-way through the year using the V2 format.
+  # For such states, we specify which update of the 2017-2018 file is in the V2
+  # format.
+  UPDATES_FOR_STATES_USING_FORMAT_V2_IN_MID_2017_18 = {
+    RI: 2
+  }.freeze
+
   # For a few states, we already placed a 2018-2019 CSV on S3 using their V1 format,
   # but they are now getting an update mid-way through the year using the V2 format.
   # For such states, we specify which update of the 2018-2019 file is in the V2
   # format.
   UPDATES_FOR_STATES_USING_FORMAT_V2_IN_MID_2018_19 = {
     DC: 2,
     DE: 2,
-    HI: 2,
     NH: 2,
     NJ: 2,
     VT: 2
   }.freeze
 
   def self.state_uses_format_v2(state_code, school_year, update)
```

Comment:
This method is starting to get complicated.  I can follow the logic because I helped design this approach.  I'm now wondering whether putting the format in the filename (`LA/2018-2019.v2.csv`) and not having multiple files per year would be easier for other engineers to maintain in the future.

Reply:
I like that idea!  Making a stronger assertion about format as part of the filename makes a lot of sense.

Reply:
It's a good point, and something I considered last week.  The main reason I decided not to was because we would have two forms of versioning in the one filename - one for the format, and one to declare that this was an incremental update for the same year - and the latter was the long-term investment while the former was a transitional measure.  Best would be to store the format version as metadata in the file, but unfortunately CSV doesn't lend itself to having additional metadata.  I've also pondered using a heuristic to determine the format version, such as the presence of certain named fields, but that also feels fragile if we want to update the format version for other reasons.

Long story short, I agree that the complexity is on the tipping point of needing a refactor.  That said, I'm hopeful that this is the end of of ingestions for the current year, and that everything gets simpler for 2019-20 and beyond when everything is in the new format.